### PR TITLE
ATO-573: Create orch token lambda

### PR DESCRIPTION
**Pull request by CarlyG55**
## What
Set up token lambda in orchestration

### CODE REVIEW

**Review by isaac-GDS — DISMISSED** [review]

File: template.yaml [hunk 1]
```diff
@@ -620,6 +635,199 @@ Resources:
...
+              - kms:ReEncrypt*
+              - kms:GenerateDataKey*
+              - kms:CreateGrant
+              - kms:DescribeKey
+            Resource: !FindInMap [ EnvironmentConfiguration, !Ref Environment, clientRegistryTableKeyArn ]
+          - Sid: AllowUserProfileTableReadAndWriteAccess
+            Effect: "Allow"
+            Action:
+              - dynamodb:BatchGetItem
+              - dynamodb:DescribeStream
```
Comment (ethanmills) [inline template.yaml:720]: Don't think we need all of these?

**Review by ethanmills — APPROVED** [review]

File: template.yaml (hunk 1 above)
Comment (CarlyG55) [reply template.yaml:720]: I just took these permissions from the terraform permissions - I can have a bit more of a think about them

File: template.yaml [hunk 2]
```diff
@@ -625,6 +640,201 @@ Resources:
...
+            Action:
+              - dynamodb:BatchGetItem
+              - dynamodb:DescribeTable
+              - dynamodb:Get*
+              - dynamodb:Query
+              - dynamodb:Scan
+            Resource: !Sub
+              - 'arn:aws:dynamodb:eu-west-2:${AccountId}:table/${AuthEnvironment}-client-registry'
+              - AccountId: !FindInMap [ EnvironmentConfiguration, !Ref Environment, authAccountId ]
+                AuthEnvironment: !FindInMap [ EnvironmentConfiguration, !Ref Environment, authEnvironment ]
```
Comment (hannah-leong) [inline template.yaml:715]: Very nit-picky but do the quotation marks need to be here? Just wondering for consistency as most of the other statements (or at least most bits of them) don't have them

**Review by hannah-leong — APPROVED** [review]

**Review by ethanmills — APPROVED** [review]

File: template.yaml (hunk 2 above)
Comment (CarlyG55) [reply template.yaml:715]: I just copied them over but I can change in both places

Comment (CarlyG55) [reply template.yaml:715]: Although I might do this later on, as I probably want to check it deploys happily and sandpit is busy today

**Pull request merged**